### ISSUE TITLE
[Bug] Storybook 7 Controls doesn't change component within iframe unless manually trigger refresh

### ISSUE THREAD

**Issue by haleksandre** [storybookjs/addon-svelte-csf#80]
### Describe the bug

Controls values doesn't reflect  on the component unless clicking manually the remount button or a full page refresh with args preset in url. 

Reproduction repo: https://github.com/haleksandre/storybook7-svelte

### Steps to reproduce the behavior

1. Run storybook (npm run storybook)
2. Go to button component
3. Change controls values
4. Component doesn't get refreshed (until manual clicking remount or full page refresh)

### Expected behavior

1. Run storybook (npm run storybook)
2. Go to button component
3. Change controls values
4. Component get refreshed realtime with controls values

### Environment

 - OS: Ubuntu
 - Node.js version: v18.12.1
 - NPM version: 8.19.2
 - Browser: Brave

**Comment by jnschrag** [comment]
@JReinhold It looks like this is the same/similar issue that I mentioned to you on Discord where changing the controls were not updating the component when using an "in between" component like in [this example](https://github.com/MiniKraken-Team/Storybook-Svelte-Named-Slots/blob/main/src/stories/migration-components/Button-Migration.svelte#L4-L13).

However, if I change it to follow this kind of pattern instead, changing the controls works:

```
<script>
  import Button from '../Button.svelte'

export let backgroundColor;
export let prefix;
export let suffix;
export let defaultSlot;
export let onClick;
// ...etc.
</script>


<Button {backgroundColor} on:click={onClick}>
  <svelte:fragment slot="prefix">
    {prefix} <!-- could add @html if html is inserted here as well -->
  </svelte:fragment>

  {defaultSlot}

  <svelte:fragment slot="suffix">
    {suffix} <!-- could add @html if html is inserted here as well -->
  </svelte:fragment>
</Button>
```

**Comment by ZackMitkin** [comment]
I've created a reproduction of this issue here in this repo. Reproduction steps are in the README.md
https://github.com/ZackMitkin/storybook-svelte-csf-vite-example

**Comment by hallefsantos** [comment]
hey @haleksandre, any updates on this?

**Comment by alexlafroscia** [comment]
I've noticed two other related issues (maybe it's actually the same issue, represented two ways) with the way that Controls work with `argTypes` in the current Svelte handling (in my case, using `beta.47`). These are _not_ present in the version of Storybook used in the reproduction that @haleksandre provided; I took that reproduction and upgraded the Storybook packages to the version I have in my own app, to validate that the issue I'm seeing in my app is present there as well (which _is_ the case)

- A `defaultValue` for an argument is not actually set as the default for an `arg` when used in the story; the story receives `undefined` as the value instead
- The control does not default to displaying the right value if you haven't explicitly set a value; for example, this `argType` configures a `defaultValue` but the control does not reflect it
- 
<img width="490" alt="CleanShot 2023-02-17 at 12 44 30@2x" src="https://user-images.githubusercontent.com/1645881/219732055-9821020c-c33e-449d-9af8-eecfbeffa0b3.png">

**Comment by ihym** [comment]
still an issue in beta.62

**Comment by jacksmethurst** [comment]
Having the same issue in (@storybook/addon-svelte-csf: 3.0.0-next.4)

**Comment by leika** [comment]
Had the same issue with controlls not refreshing the component/story after args changed. Created a PR that fixed this for me https://github.com/storybookjs/addon-svelte-csf/pull/99

Essentially, the Wrapper Component would need to react on $$props change and recreate the Story Component. No iframe refresh needed in those cases.

**Comment by mantismamita** [comment]
> Essentially, the Wrapper Component would need to react on $$props change and recreate the Story Component. No iframe refresh needed in those cases.

It would amazing if the iframe didn't refresh as it triggers an additional and unnecessary onMount which can complicate testing.

**Comment by leika** [comment]
The other fix would be to wrap the args context into a writeable and make sure Template and Story components are correctly react on changes to tis Store

**Comment by mantismamita** [comment]
> The other fix would be to wrap the args context into a writeable and make sure Template and Story components are correctly react on changes to tis Store
 
 IMO, it should reflect the behaviour of the component outside of an iframe. The store option seems like a good idea.

**Comment by leika** [comment]
Agree, need to look into this. We'd need stores for args and context in this case (otherwise changes to context.globals would equally be lost, which is important in case someone uses global toolbar buttons for theme switching or similar)

**Comment by leika** [comment]
I did update the PR now, using stores and also a clean context at render time. I think that's very close to vanilla svelte behaviour.

**Comment by bjornet** [comment]
I also encountered this issue – where controls do not update the canvas on change. I tried @leika:s solution. It worked well, looking forward for this to get into main 👍🏽

**Comment by iTechTomekK** [comment]
Yes, I am still waiting for it too - is there a reason why it's still not fixed ? Does it need testing power or something ?

**Comment by shilman** [comment]
:rocket: Issue was released in [`v3.0.2`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v3.0.2) :rocket:

**Comment by miedzikd** [comment]
Hey! Does anyone still experience this issue? After changing the component parameters, the component is not unmounted, so if any variables inside the component rely on the attribute value, they won't be refreshed...

I have prepared a minimal reproduction ( https://stackblitz.com/edit/storybook-svelte-bug?file=README.md,assets%2Fsvelte%2Fui%2Fatoms%2FButton%2FButton.svelte ). Please note that the button component has 2 parameters:
- The first one, `color` does not refresh because the `classes` variable is built based on it. In previous versions of Storybook, this also used to refresh. 
- The change of the second parameter, `disabled`" is immediately visible because it is directly used in the HTML code.

So is this how Storybook Svelte works, and is it the intended behavior? Do I need to change all the components? Or is this considered incorrect behavior?
